CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -154,37 +154,43 @@ function Staff:tick()
   self:updateSpeed()
 end
 
-function Staff:checkIfWaitedTooLong()
+function Staff:checkIfWaitedTooLongForRaise()
   if self.quitting_in then
     self.quitting_in = self.quitting_in - 1
-    if self.quitting_in < 0 then
+
+    local is_waiting_time_is_up = self.quitting_in < 0
+    if is_waiting_time_is_up then
```

Comment:
We can also return early to remove one nesting.
```lua
if not self.quitting_in < 0 then return end
--[[or]]
if self.quitting_in >= 0 then return end
```

Reply:
> The local isn't needed here imo, `self.quitting_in < 0` is self-explanatory. I wonder if it should be `<` or `<=`

It was probably pretty clear, but I thought it could be made even clearer.
`is_waiting_time_is_up` is super obvious. 
I think used `<` and not `<=` because it want to stop on `0` which was when we just entered this function call. It seems more logical than stopping at 1.

Reply:
> We can also return early to remove one nesting.
> 
> ```lua
> if not self.quitting_in < 0 then return end
> --[[or]]
> if self.quitting_in >= 0 then return end
> ```

Sorry, didn't get that.

Reply:
By flipping your if condition, you can exit the function early because we no longer need to do anything more.

Reply:
> By flipping your if condition, you can exit the function early because we no longer need to do anything more.

Oh, yeah, I got it now. Thanks. It was not mine, it was there before. But yes, I agree. Changed nesting level.